autoreject/autoreject.py

```diff
@@ -530,30 +482,132 @@ def _interpolate_bad_epochs(self, epochs, verbose='progressbar'):
+                                   n_consensus) + 1
+            bad_epochs_idx = sorted_epoch_idx[:n_epochs_drop]
+        else:
+            n_epochs_drop = 0
+            bad_epochs_idx = []
+
+        return bad_epochs_idx, sorted_epoch_idx, n_epochs_drop
+
+    def _annotate_epochs(self, threshes, epochs):
+        """Get essential annotations for epochs given thresholds."""
+        ch_type = _get_ch_type_from_picks(self.picks, epochs.info)[0]
+
+        drop_log, bad_sensor_counts = self._vote_bad_epochs(epochs)
+
+        bad_channels, fix_log = self._get_epochs_interpolation(
+            epochs, drop_log=drop_log, ch_type=ch_type)
+
+        (bad_epochs_idx, sorted_epoch_idx,
+         n_epochs_drop) = self._get_bad_epochs(
+             bad_sensor_counts, ch_type=ch_type)
+
+        bad_epochs_idx = np.sort(bad_epochs_idx)
+        good_epochs_idx = np.setdiff1d(np.arange(len(epochs)),
+                                       bad_epochs_idx)
+
+        return (drop_log, bad_sensor_counts, bad_channels, fix_log,
+                bad_epochs_idx, good_epochs_idx)
+
+    def fit(self, epochs):
+        """Compute the thresholds.
+
+        Parameters
+        ----------
+        epochs : instance of mne.Epochs
+            The epochs object from which the channel-level thresholds are
+            estimated.
+        """
+        self.picks = _handle_picks(info=epochs.info, picks=self.picks)
+        _check_data(epochs, picks=self.picks, verbose=self.verbose,
+                    ch_constraint='single_channel_type')
+        self.threshes_ = self.thresh_func(
+            epochs.copy(), picks=self.picks, verbose=self.verbose)
+
+        (drop_log, bad_sensor_counts, bad_channels, fix_log,
+         bad_epochs_idx, good_epochs_idx) = self._annotate_epochs(
+             threshes=self.threshes_, epochs=epochs)
+
+        self.drop_log_ = drop_log
+        self.fix_log_ = fix_log
+        self.bad_sensor_counts_ = bad_sensor_counts
+        self.bad_channels_ = bad_channels
+        self.bad_epochs_idx_ = bad_epochs_idx
+        self.good_epochs_idx_ = good_epochs_idx
+
+        epochs_copy = epochs.copy()
+        self._interpolate_bad_epochs(epochs_copy, bad_channels=bad_channels,
+                                     verbose=self.verbose)
+        self.mean_ = _slicemean(epochs_copy.get_data(),
+                                good_epochs_idx, axis=0)
+        del epochs_copy  # I can't wait for garbage collection.
+        return self
+
+    def transform(self, epochs):
+        """Fix and find the bad epochs.
+
+        Parameters
+        ----------
+        epochs : instance of mne.Epochs
+            The epochs object for which bad epochs must be found.
+        """
+        _check_data(epochs, picks=self.picks, verbose=self.verbose,
+                    ch_constraint='single_channel_type')
+
+        (drop_log, bad_sensor_counts, bad_channels, fix_log,
+         bad_epochs_idx, good_epochs_idx) = self._annotate_epochs(
```

Comment:
a lot of these returned variables are not used. Maybe we don't need to return them?